lib/MusicBrainz/Server/Entity/CDTOC.pm

```diff
@@ -143,10 +131,18 @@ with 'MusicBrainz::Server::Entity::Role::TOC';
 around TO_JSON => sub {
     my ($orig, $self) = @_;
 
-    my $json = $self->$orig;
-    $json->{discid} = $self->discid;
-
-    return $json;
+    my $data = {
+        %{ $self->$orig },
+        discid          => $self->discid,
+        freedb_id       => $self->freedb_id,
+        leadout_offset  => $self->leadout_offset,
+        length          => $self->length,
+        track_count     => $self->track_count,
+        track_details   => $self->track_details,
+        track_offset    => $self->track_offset,
```

Comment:
Should probably add `0 + ...` to all the numeric ones to ensure Perl doesn't encode them as strings.  (Not sure if any can be `undef` though.)

Reply:
In theory they can't, by the definition (if some of our data is bad, that I can't say)